include/circt/Dialect/Calyx/CalyxLoweringUtils.h

```diff
@@ -424,6 +424,10 @@ class ComponentLoweringStateInterface {
     Block *body = component.getBodyBlock();
     builder.setInsertionPoint(body, body->begin());
     auto name = TLibraryOp::getOperationName().split(".").second;
+    if (std::is_same<calyx::AddFNOp, TLibraryOp>::value)
+      name = "std_addFN";
```

Comment:
Nit: isn't the better way to do this is remove the `.ieee754` prefix for any `F` operations?

Reply:
The prefix is `calyx.ieee754.add`/`calyx.ieee754.mul`, removing `ieee754` will simply become `add`/`mul`; but we want `std_addFN`/`std_mulFN` after emitting to Calyx native compiler.

Reply:
More concretely, I think there is a better way to do this than manually write a new `if` statement for each floating point operation. It'd be nice to avoid this every time:

```
if (std::is_same<calyx::<MyNewFloatingPointOp, TLibraryOP>::value)
  name = "std_MyNewFloatingPointOp";
```

Ideally, since we're already distinguishing them without a `_std` prefix, we could infer this is some floating point operation, and then check what standard it conforms too.

```cpp
/// Returns the technical floating point this operation conforms to, e.g., "ieee754"
/// (This should probably be an enum, which is subsequently converted to a string.)
std::string standard() { ... }

if (auto fOp = dyn_cast<FloatingPointOp>(op)):
  name.remove(llvm::join(".", fOp.standard()));
  name.prepend("std_");
```

(I'm not saying you should spend a lot of time doing this, it is a mere suggestion. In general, I do think the conforming technical standard of a floating point operation should be a field of the op so it can referenced during pattern rewrites.)

Reply:
I've thought about this, I don't think it's worth it now.
In the foreseeable future, only IEEE754 floating point is integrate; and moreover, only Add and Mul are present in the Berkeley HardFloat.
My opinion is we can address future needs and refactor the code when there is an actual, concrete need.
But I totally get what you are saying.

Reply:
Sorry, let me take it back.
When doing some research on the floating point comparison issue https://github.com/llvm/circt/issues/7821, I found Berkeley HardFloat has division, square root, and comparisons as well.
So I definitely need to use a better design to incorporate all of these `calyx::FloatingPointOp`s, as you suggested.